src/App.tsx

```diff
@@ -21,75 +21,139 @@ enum SortType {
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+  goods: string[],
 };
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const visibleGoods = [...this.state.goods];
+    const { isReversed, sortType } = this.state;
+
+    visibleGoods.sort((a, b): SortType => {
+      switch (sortType) {
+        case SortType.LENGTH:
+          return a.length - b.length;
+
+        case SortType.ALPHABET:
+          return a.localeCompare(b);
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <>
+        <div className="section content">
+          <div className="buttons">
+            <button
+              type="button"
+              className={`button is-info ${this.state.sortType !== SortType.ALPHABET ? 'is-light' : ''}`}
+              onClick={
+                () => {
+                  this.setState({
+                    sortType: SortType.ALPHABET,
+                  });
+                }
+              }
+            >
+              Sort alphabetically
+            </button>
+
+            <button
+              type="button"
+              className={`button is-success ${this.state.sortType !== SortType.LENGTH ? 'is-light' : ''}`}
+              onClick={
+                (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+                  this.setState({
+                    sortType: SortType.LENGTH,
+                  });
+                  event.currentTarget.classList.remove('is-light');
+                }
+              }
+            >
+              Sort by length
+            </button>
+
+            <button
+              type="button"
+              className={`button is-warning ${this.state.isReversed !== true ? 'is-light' : ''}`}
+              onClick={
+                () => {
+                  this.setState((state) => ({
+                    isReversed: !state.isReversed,
+                  }));
+                }
+              }
+            >
+              Reverse
+            </button>
+
+            {
+              (visibleGoods[0] !== this.state.goods[0]) && (
+                <button
+                  type="button"
+                  className="button is-danger is-light"
+                  onClick={() => {
+                    this.setState(() => ({
+                      sortType: SortType.NONE,
+                      isReversed: false,
+                    }));
+                  }}
+                >
+                  Reset
+                </button>
+              )
+            }
+          </div>
+
+          <ul>
+            <ul>
```

Comment:
Do you really need to have two nested `ul` tags here?

Reply:
yes, you don't have to persist both, kill one of these

Reply:
what about this one?)